pkg/watchdog/watchdog.go

```diff
@@ -68,18 +69,15 @@ func WatchdogFileRemove(baseDir string, vmi *v1.VirtualMachineInstance) error {
 	return diskutils.RemoveFilesIfExist(file)
 }
 
-func WatchdogFileUpdate(watchdogFile string, uid string) error {
+func WatchdogFileUpdate(watchdogFile string, uid string) (err error) {
```

Comment:
nit: you don't need a named return values

Reply:
Yes, If we want the error from `defer util.CloseIOAndCheckErr()` to be returned from this function.

Reply:
It may be a little confusing. Line 73 does not define a new variable called `err`, it only defines new `f`, but uses already existing `err`, so there is no shadowing. The `err` pointer in `defer` statement refers to the named return value.

Reply:
Ok, then I'll use it also in all the return statements `return err` -> `return`. Otherwise, it looks good to me

Reply:
it would signal that you are using a named return value and that the error might be overwritten in the defer statement. But, it's ok if you still want to keep it :)